project/src/main/java/com/google/sps/servlets/UserProfileEventsServlet.java

```diff
@@ -57,7 +59,13 @@ private String getEventsJSONByEmail(String eventType, String userEmail) {
             SpannerTasks.getEventsVolunteeringByEmail(userEmail);
         eventJSON = CommonUtils.convertToJson(eventsVolunteering);
         break;
-      // TO DO: add case statements for hosting and participating with retrieval of data
+      case PARTICIPATING:
+        Set<Event> eventsParticipating =
+            SpannerTasks.getEventsParticipatingByEmail(userEmail);
+        eventJSON = CommonUtils.convertToJson(eventsParticipating);
+        break;
```

Comment:
Please try to use the most updated version of [1/X] from code review it was suggested to use a return here rather than using the variable eventsJSON.